### PR TITLE
feat: dropdown with avatar

### PR DESCRIPTION
**Pull request by augustobmoura**
[FX-3129]

### Description

Add avatar option to MenuItem and consequently to Dropdown

### How to test

- Check demo and Happo tests

### Screenshots

image

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3129]: https://toptal-core.atlassian.net/browse/FX-3129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso/src/MenuItem/MenuItem.tsx [hunk 1]
```diff
@@ -129,52 +133,51 @@ export const MenuItem: OverridableComponent<Props> = forwardRef<
...
-        <Container
-          ref={anchorRef}
-          flex
-          direction='column'
-          className={classes.content}
-        >
-          <Container flex alignItems='center'>
-            {isIconWrapperVisible && (
+        <Container ref={anchorRef} flex direction='row'>
+          {avatar && <Container className={classes.avatar}>{avatar}</Container>}
```
Comment (ozgurkececioglu) [inline packages/picasso/src/MenuItem/MenuItem.tsx:137]: I would prefer using Container prop
```suggestion
          {avatar && <Container right='xsmall'>{avatar}</Container>}
```

**Review by ozgurkececioglu — COMMENTED** [review]
The new layout for MenuItem content doesn't look right 🤔 

| before | after |
| ----- | ----- |
|Screenshot 2022-11-08 at 10 39 01|
Screenshot 2022-11-08 at 10 39 05|

File: packages/picasso/src/MenuItem/MenuItem.tsx [hunk 2]
```diff
@@ -58,6 +59,8 @@ export interface Props extends BaseProps, TextLabelProps, MenuItemAttributes {
   description?: ReactNode
   /** Render an `<Icon />` */
   icon?: ReactElement
+  /** Render an Avatar */
```
Comment (LashaJini) [inline packages/picasso/src/MenuItem/MenuItem.tsx:62]: nit (just a consistency)
```suggestion
  /** Render an <Avatar /> */
```

**Review by LashaJini — APPROVED** [review]
Code looks good. Thank you Augusto :+1:

File: packages/picasso/src/MenuItem/MenuItem.tsx [hunk 3]
```diff
@@ -103,7 +107,7 @@ export const MenuItem: OverridableComponent<Props> = forwardRef<
     onMouseEnter,
   })
   const isLink = as === Link && rest.href
-  const isIconWrapperVisible = checkmarked !== undefined || icon
+  const isIconWrapperVisible = checkmarked !== undefined || icon || !avatar
```
Comment (TomasSlama) [inline packages/picasso/src/MenuItem/MenuItem.tsx:110]:
```suggestion
  const isIconWrapperVisible = checkmarked !== undefined || icon || avatar
```

File: packages/picasso/src/MenuItem/MenuItem.tsx (hunk 1 above)
Comment (augustobmoura) [reply packages/picasso/src/MenuItem/MenuItem.tsx:142]: Nice catch, I forgot about `right`

File: packages/picasso/src/MenuItem/MenuItem.tsx (hunk 2 above)
Comment (augustobmoura) [reply packages/picasso/src/MenuItem/MenuItem.tsx]: Done :+1:

**Review by ozgurkececioglu — APPROVED** [review]
tested a lot and approving with pleasure. Good job 👏 🙇‍♂️

**Review by TomasSlama — APPROVED** [review]
Tested in temploy, looks great, thank you Augusto

**Pull request merged**